source/module_io/test/parameter_pool_set_test.cpp

```diff
@@ -0,0 +1,95 @@
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <map>
+#include <sstream>
+#include <utility>
+#include <vector>
+
+#include "gtest/gtest.h"
+#include "module_io/parameter_pool.h"
+
+/**
+ * @brief test func:input_parameters_get() and input_parameters_set() in parameter_pool.h
+ */
+TEST(InputParameterArgsTest, InputTest)
```

Comment:
Do you expected other users use input module follow this UT? 
In this example, there is a risk usage of pointer cast:
`
inputparaBool.set((void*)&b1);
ModuleIO::input_parameters["bool"] = inputparaBool;
bool b2 = *static_cast<bool*>(ModuleIO::input_parameters["bool"].get());
`
A bool* -> void* ->bool* , if user misuse it as bool* ->void* ->double*, any error will be throwed?

Reply:
This test code is used to test the correctness of parameter_input module parameter passing. For users, these codes do not need to be written, and the code logic has been written in the current module.
  parameter_input acts as a bridge between input files and input.h. It configures customizable configuration files (including default values and user-input values) to input.h. A map with parameters value and types is then generated -- ModuleIO::input_parameters["sth"]. In use (new requirements: printing all variables, for example) it is only necessary to know the correct type of sth and use the corresponding pointer.

Reply:
This test code is used to test the correctness of parameter_input module parameter passing. For users, these codes do not need to be written, and the code logic has been written in the current module.
parameter_input acts as a bridge between input files and input.h. It configures customizable configuration files (including default values and user-input values) to input.h. A map with parameters value and types is then generated -- ModuleIO::input_parameters["sth"]. In use (new requirements: printing all variables, for example) it is only necessary to know the correct type of sth and use the corresponding pointer.